process_pub_database/get_data_cpdl.py

```diff
@@ -8,32 +8,108 @@
 import time
 import unicodedata
 
+from utils import *
+
+
+def find_composer(bs_file):
+	compy = bs_file.find_all('b',text = re.compile('Composer|Editors'))[0].find_next_siblings('a')[0]
+	compy_name = compy['title']
+	compy_name = unicodedata.normalize('NFKC', compy_name).strip()
+	composer = {'Name' : compy_name, 
+	'url':'https://www.cpdl.org'+compy['href']}
+	return composer
+
+def process_composer(comp_str):
+
+	composer_file = bs(comp_str, features="lxml")
+
+	dict_comp = {}
+
+	if 'Biography' in comp_str:
+		biop = composer_file.find_all('b',text = re.compile('Biography'))[0].parent
+		bio = biop.getText().rstrip()+biop.find_next_siblings()[0].getText().strip()
+		bio = bio.replace('Biography','').replace(':','').strip()
+	elif 'Wikipedia article' in comp_str:
+		bio = composer_file.find_all(text = re.compile('Wikipedia article'))[0].parent.find_previous_siblings('p')[0].getText().strip()
+	if 'Wikipedia article' in comp_str:
+
+		wiki_link = composer_file.find_all(text = re.compile('Wikipedia article'))[0].parent.find('a')['href']
+	else: 
+		wiki_link = ''
+
+	if 'Born' in comp_str:
+		born = composer_file.find_all(text = re.compile('Born'))[0].parent.next_sibling.string.strip()
+	else:
+		born = 'Unknown'
+	if 'Died' in comp_str:
+		died = composer_file.find_all(text = re.compile('Died'))[0].parent.next_sibling.string.strip()
+	else:
+		died = 'Unknown'
+
+	dict_comp['Biography'] = bio
+	dict_comp['Wikipedia'] = wiki_link
+	dict_comp['Born'] = born
+	dict_comp['Died'] = died
+
+	return dict_comp
+
+def find_name(title_file):
```

Comment:
these methods are good - it's a good idea to split the code that selects each value into a separate function. This makes it easier to test a specific field, and to centralise error checking for this field.
I'd recommend to do the same in the imslp scraper too, even if it's just selecting a single value from the html document. It should make testing and extending the file easier in the future.